src/test/java/org/opensearch/security/dlic/dlsfls/DlsTermLookupQueryTest.java

```diff
@@ -0,0 +1,612 @@
+
+		// user has 1337 and 42, document has 1337
+		GetResponse searchResponse = executeGet("tlqdocuments", "1", "tlq_1337_42", "password");
+		Assert.assertTrue(searchResponse != null);
+		Assert.assertTrue(searchResponse.isExists());
+		assertAccessCodesMatch(searchResponse.getSourceAsMap(), "access_codes", new Integer[] { 1337, 42 });
+
+		// user has 1337 and 42, document has 42
+		searchResponse = executeGet("tlqdocuments", "2", "tlq_1337_42", "password");
+		Assert.assertTrue(searchResponse != null);
+		Assert.assertTrue(searchResponse.isExists());
+		assertAccessCodesMatch(searchResponse.getSourceAsMap(), "access_codes", new Integer[] { 1337, 42 });
+
+		// user has 1337 and 42, document has 42 and 1337
+		searchResponse = executeGet("tlqdocuments", "3", "tlq_1337_42", "password");
+		Assert.assertTrue(searchResponse != null);
+		Assert.assertTrue(searchResponse.isExists());
+		assertAccessCodesMatch(searchResponse.getSourceAsMap(), "access_codes", new Integer[] { 1337, 42 });
+
+		// user has 1337 and 42, document has no access codes, not visible
+		searchResponse = executeGet("tlqdocuments", "16", "tlq_1337_42", "password");
+		Assert.assertFalse(searchResponse.isExists());
+
+		// user has 1337 and 42, document has 12345, not visible
+		searchResponse = executeGet("tlqdocuments", "17", "tlq_1337_42", "password");
+		Assert.assertFalse(searchResponse.isExists());
+
+		// user has 1337 and 42, document has 12345 and 6789, not visible
+		searchResponse = executeGet("tlqdocuments", "18", "tlq_1337_42", "password");
+		Assert.assertFalse(searchResponse.isExists());
+
+	}
+
+	@Test
+	public void testGet_TlqDummyIndex_1337() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+
+		// no restrictions on this index
+		GetResponse searchResponse = executeGet("tlqdummy", "101", "tlq_1337", "password");
+		Assert.assertTrue(searchResponse != null);
+		Assert.assertTrue(searchResponse.isExists());
+
+		searchResponse = executeGet("tlqdummy", "102", "tlq_1337", "password");
+		Assert.assertTrue(searchResponse != null);
+		Assert.assertTrue(searchResponse.isExists());
+
+	}
+
+	@Test
+	public void testGet_UserAccessCodesIndex_1337() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+
+		// we expect a security exception here, user has no direct access to
+		// user_access_codes index
+		HttpResponse response = rh.executeGetRequest("/user_access_codes/_doc/tlq_1337",
+				encodeBasicHeader("tlq_1337", "password"));
+		Assert.assertEquals(403, response.getStatusCode());
+		Assert.assertEquals("Forbidden", response.getStatusReason());
```

Comment:
Is this necessary given the line above it?

Reply:
True, removed the latter `Assert` statement